packages/worker/src/worker.ts

```diff
@@ -1596,10 +1596,16 @@ export class Worker {
     }
     this.state = 'RUNNING';
 
-    const shutdownCallback = () => this.shutdown();
-    Runtime.instance().registerShutdownSignalCallback(shutdownCallback);
-
     try {
+      const shutdownCallback = () => this.shutdown();
```

Comment:
There's some more teardown happening below that's missed here.

Note that below the worker transitions to "STOPPED" state before it finalizes shutdown.
We could change that so we transition only after finalizeShutdown is called below and reuse the same teardown code.